### ISSUE TITLE
Develop caching mechanisms

### ISSUE THREAD

**Issue by abinashmeher999** [abinashmeher999/pymediawiki#8]
This would aim to minimize calls to the API for repetitive requests. The idea is to use the information which has already been obtained from the API for a second query of the same page. There are many more concerns like how long before the cache expires etc which can be further discussed here if anyone is interested.

**Comment by theskcd** [comment]
Hi all,
I think we can use either redis/mecahce to implement the same. I would like to discuss more about about this :smile:

**Comment by abinashmeher999** [comment]
I just had a look at redis python package. It requires a redis server to be running. If we look at the use case of the package, personally I feel the user isn't going to setup a redis server to use the feature. Correct me if I am wrong but isn't redis mostly used on the server side? The one we are looking at is a client side one. 

We are looking at just saving a few repetitive requests. Is setting up redis worth it? IMO a naive one would work too. What do you think?

**Comment by theskcd** [comment]
Well, You are right when you say that. I think instead we can have inter-program communication modules set up so that there is some in-memory caching mechanism with fixed "memory-units" (so that it dosent end up eating a lot of system memory) where we can cache and query for the data. 
What do you say to that?

**Comment by abinashmeher999** [comment]
That sounds good! :+1:

**Comment by theskcd** [comment]
I will start working on something then.. :+1:  , so should i first fork the repo and make my commits over there?

**Comment by abinashmeher999** [comment]
@sandeepkumarpani888 Yes. Go ahead.

**Comment by abinashmeher999** [comment]
@sandeepkumarpani888 What are your thoughts on using a bloom filter for checking if something should be cached?
P.S. See the first example in the 'Examples' section https://en.wikipedia.org/wiki/Bloom_filter

**Comment by theskcd** [comment]
Yeah, I think we can work with this .. but there must be already be a library for the same. :)

**Comment by abinashmeher999** [comment]
There is. [pybloom](https://pypi.python.org/pypi/pybloom/1.0.3).

**Comment by theskcd** [comment]
Nice !

**Comment by abinashmeher999** [comment]
Just discovered this gem *pun not intended* https://pypi.python.org/pypi/cached-property and @ayuhsya has volunteered to integrate this over the weekend. :smile:
